contracts/deploy/allowListPaymaster.ts

```diff
@@ -14,51 +11,52 @@ const PRIVATE_KEY = process.env.WALLET_PRIVATE_KEY || "";
 if (!PRIVATE_KEY)
   throw "⛔️ Private key not detected! Add it to the .env file!";
 
-export default async function (hre: HardhatRuntimeEnvironment) {
-  console.log(`Running deploy script for the AllowlistPaymaster contract...`);
-  const provider = new Provider((hre.network.config as HttpNetworkUserConfig).url);
+async function main() {
+  const artifact = "AllowlistPaymaster";
+  console.log(
+    `Running script to deploy ${artifact} contract on ${hre.network.name}`,
+  );
 
-  // The wallet that will deploy the token and the paymaster
-  // It is assumed that this wallet already has sufficient funds on zkSync
-  const wallet = new Wallet(PRIVATE_KEY);
-  const deployer = new Deployer(hre, wallet);
+  // Retrieve signers
+  const [deployer] = await hre.ethers.getSigners();
 
   // Deploying the paymaster
-  const paymasterArtifact = await deployer.loadArtifact("AllowlistPaymaster");
-  const deploymentFee = await deployer.estimateDeployFee(paymasterArtifact, []);
-  const parsedFee = ethers.utils.formatEther(deploymentFee.toString());
-  console.log(`The deployment is estimated to cost ${parsedFee} ETH`);
-  // Deploy the contract
-  const paymaster = await deployer.deploy(paymasterArtifact, []);
-  console.log(`Paymaster address: ${paymaster.address}`);
-  console.log(`Contract owner added to allow list: ${wallet.address}`);
+  const paymaster = await deployContract(artifact, []);
+  const paymasterAddress = await paymaster.getAddress();
+  console.log(`Paymaster address: ${paymasterAddress}`);
+  console.log(`Contract owner added to allow list: ${deployer.address}`);
 
   console.log("Funding paymaster with ETH");
   // Supplying paymaster with ETH
-  await (
-    await deployer.zkWallet.sendTransaction({
-      to: paymaster.address,
-      value: ethers.utils.parseEther("0.005"),
-    })
-  ).wait();
-
-  let paymasterBalance = await provider.getBalance(paymaster.address);
-  console.log(`Paymaster ETH balance is now ${paymasterBalance.toString()}`);
+  await fundAccount(deployer, paymasterAddress, "0.005");
 
-  // Verify contract programmatically
-  //
-  // Contract MUST be fully qualified name (e.g. path/sourceName:contractName)
-  const contractFullyQualifedName =
-    "contracts/paymasters/AllowlistPaymaster.sol:AllowlistPaymaster";
-  const verificationId = await hre.run("verify:verify", {
-    address: paymaster.address,
-    contract: contractFullyQualifedName,
-    constructorArguments: [],
-    bytecode: paymasterArtifact.bytecode,
-  });
-  console.log(
-    `${contractFullyQualifedName} verified! VerificationId: ${verificationId}`,
+  const paymasterBalance = await hre.ethers.provider.getBalance(
+    paymasterAddress,
   );
+  console.log(`Paymaster ETH balance is now ${paymasterBalance.toString()}`);
+  if (hre.network.name.includes("ZKsyncEra")) {
```

Comment:
Perhaps we change this to check if `hre.network.config.verifyURL` exists?

Reply:
the plugin injects `verifyURL` with testnet details by default if you don't provide it 🫤